lib/datadog/core/configuration/option.rb

```diff
@@ -8,7 +8,7 @@ module Configuration
       # Represents an instance of an integration configuration option
       # @public_api
       class Option
-        attr_reader :definition
+        attr_reader :definition, :precedence_set
```

Comment:
What is `precedence_set`?

Reply:
This is the effective precedence of the value this option is currently set to.
For example, if this option was never set, the precedence will be "default".
If the same option is then set by programmatic configuration (`Datadog.configure{}`), it will change to "programmatic".
Then, again, if it is over, written by remote configuration, it will change to "remote configuration".

Now, the interesting part is that same option is then again remote configuration by programmatic configuration, it will not change value, given remote configuration has a higher precedence.

Let me know if this makes sense. Also, I expanded the in-line documentation of this class regarding precedence.

On another topic, it probably makes sense to simply rename me to `precedence`, what do you think? I would probably do that in a separate PR as, it will touch many files and related to this PR, but it's a very simple rename to do as a follow up.

Reply:
Gotcha. Was just curious. No strong opinions at this point, so if you're confident on this, I'm interested to see how it works.